### PR TITLE
add home/delete endpoint

### PR DESCRIPTION
**Pull request by isiktashamza**
I added a simple home/delete endpoint and tested it with postman. Any review will be welcomed

### CODE REVIEW

**Review by EnesToptas — COMMENTED** [review]
Is home the right path for this to be done? I believe profile might be better, also posts of that user or their contributions to other projects should be deleted.

File: backend/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "http": "^0.0.1-security",
     "https": "^1.0.0",
     "jsonwebtoken": "^8.5.1",
+    "loader": "^2.1.1",
```
Comment (EnesToptas) [inline backend/package.json:22]: what is loader used for?

**Comment by isiktashamza** [comment]
> Is home the right path for this to be done? I believe profile might be better, also posts of that user or their contributions to other projects should be deleted.

I was considering auth path as the best option but auth endpoints do not require authorization token and taking userId as a parameter or as a field in the body did not make sense to me. And I did not want to create a new controller for this purpose. Therefore, I wrote it in home_controller.

File: backend/package.json (hunk 1 above)
Comment (isiktashamza) [reply backend/package.json:22]: When I ran the program it gave MODULE_NOT_FOUND error. I installed that module. But now it seems unnecessary. We can discard this line.

**Comment by hazerb** [comment]
By the way no need to write extra functions to delete users projects, or other contributions etc. Foreign key constraints do deletion automatically. Just need to check if all foreign key constraints correctly constructed in db.js file.

**Review by hazerb — COMMENTED** [review]
Other than mentioned things,  code seems okay.

**Comment by isiktashamza** [comment]
> By the way no need to write extra functions to delete users projects, or other contributions etc. Foreign key constraints do deletion automatically. Just need to check if all foreign key constraints correctly constructed in db.js file.

I checked them. Follows and projects of the deleted user are deleted, too.

**Comment by EnesToptas** [comment]
> > Is home the right path for this to be done? I believe profile might be better, also posts of that user or their contributions to other projects should be deleted.
> 
> I was considering auth path as the best option but auth endpoints do not require authorization token and taking userId as a parameter or as a field in the body did not make sense to me. And I did not want to create a new controller for this purpose. Therefore, I wrote it in home_controller.

Yes you are right, well done, I think we can merge it. But still kinda feel profile would have been better :D.

**Comment by EnesToptas** [comment]
> > By the way no need to write extra functions to delete users projects, or other contributions etc. Foreign key constraints do deletion automatically. Just need to check if all foreign key constraints correctly constructed in db.js file.
> 
> I checked them. Follows and projects of the deleted user are deleted, too.

Are they also deleted from contributors table?

**Comment by isiktashamza** [comment]
> > > Is home the right path for this to be done? I believe profile might be better, also posts of that user or their contributions to other projects should be deleted.
> > 
> > 
> > I was considering auth path as the best option but auth endpoints do not require authorization token and taking userId as a parameter or as a field in the body did not make sense to me. And I did not want to create a new controller for this purpose. Therefore, I wrote it in home_controller.
> 
> Yes you are right, well done, I think we can merge it. But still kinda feel profile would have been better :D.

It is 2-minutes job. If you wish, I can move it :)

**Comment by EnesToptas** [comment]
> > > > Is home the right path for this to be done? I believe profile might be better, also posts of that user or their contributions to other projects should be deleted.
> > > 
> > > 
> > > I was considering auth path as the best option but auth endpoints do not require authorization token and taking userId as a parameter or as a field in the body did not make sense to me. And I did not want to create a new controller for this purpose. Therefore, I wrote it in home_controller.
> > 
> > 
> > Yes you are right, well done, I think we can merge it. But still kinda feel profile would have been better :D.
> 
> It is 2-minutes job. If you wish, I can move it :)

Nah it's ok :D 👯‍♂️

**Comment by isiktashamza** [comment]
> > > By the way no need to write extra functions to delete users projects, or other contributions etc. Foreign key constraints do deletion automatically. Just need to check if all foreign key constraints correctly constructed in db.js file.
> > 
> > 
> > I checked them. Follows and projects of the deleted user are deleted, too.
> 
> Are they also deleted from contributors table?

I checked other tables. When a user is deleted, related ups and collaborator information are not deleted. It should be fixed.

**Review by EnesToptas — APPROVED** [review]
Seems fine, merging.

**Pull request merged**